edivorce/apps/core/redis.py

```diff
@@ -0,0 +1,100 @@
+"""
+Redis storages backend to help store binary data.
+"""
+import base64
+import io
+import redis
+import uuid
+import re
+
+from shutil import copyfileobj
+from tempfile import SpooledTemporaryFile
+
+from django.conf import settings
+from django.core.files.base import File
+from django.core.files.storage import Storage
+from django.utils.deconstruct import deconstructible
+
+EX_EXPIRY = 60*24*7  # 1 week expiry
+
+
+def generate_unique_filename(instance, filename):
+    return '{}_{}'.format(uuid.uuid4(), re.sub('[^0-9a-zA-Z]+', '_', filename))
+
+
+class RedisFile(File):
+
+    def __init__(self, name, storage):
+        self.name = name
+        self._storage = storage
+        self._file = None
+
+    def _get_file(self):
+        if self._file is None:
+            self._file = SpooledTemporaryFile()
+
+            # get from redis
+            content = self._storage.client.get(self.name)
+
+            # stored as base64 .. decode
+            content = base64.b64decode(content)
+
+            with io.BytesIO(content) as file_content:
+                copyfileobj(file_content, self._file)
+
+            self._file.seek(0)
+        return self._file
+
+    def _set_file(self, value):
+        self._file = value
+
+    file = property(_get_file, _set_file)
+
+
+@deconstructible
+class RedisStorage(Storage):
+    def __init__(self):
+        self.client = redis.Redis(
+            host=settings.REDIS_HOST,
+            port=settings.REDIS_PORT,
+            db=settings.REDIS_DB,
+            password=settings.REDIS_PASSWORD
+        )
+
+    def _full_key_name(self, name):
```

Comment:
What is this for?

Reply:
That was initially to account for file path when generating the file name .. normally needed for other storage solutions. Just left it in in case we change how the key names will be formatted prior to insertion into redis.